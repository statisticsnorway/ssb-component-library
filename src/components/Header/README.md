Header
========

> A wrapper for displaying a header

### Usage
The component needs to be filled with content.

#### HTML

```html
<div class="ssb-header-wrapper">
<!-- Fill with content, as below -->
    <div class="global-links" style="float: right; margin-bottom: 12px; margin-top: 10px;">
        <a class="ssb-link" href=" " target="" rel=""><span class="link-text">top-item 3</span></a>
        <a class="ssb-link" href=" " target="" rel=""><span class="link-text">top-item 2</span></a>
        <a class="ssb-link" href=" " target="" rel=""><span class="link-text">top-item 1</span></a>
    </div>
    <div class="top-row flex-row justify-space-between flex-wrap" style="width: 100%;">
        {insert SSB logo with size h:44px w:248px}
        <div class="searchfield" style="width: 285px; align-self: flex-end;">
            <div class="ssb-input ">
                <div class="input-wrapper">
                    <input id="search-input" type="text" placeholder="Search text" aria-label="Input field Search" class=" with-icon" value="">
                    <button class="icon-wrapper search-icon">
                        <i>{feather.search 18px}</i>
                    </button>
                </div>
            </div>
        </div>
    </div>
    <div class="header-content" style="margin-bottom: 20px; margin-top: 14px;">
        <div class="ssb-tabs">
            <button class="navigation-item "><span>Category 1</span></button>
            <button class="navigation-item "><span>Category 2</span></button>
            <button class="navigation-item "><span>Category 3</span></button>
            <button class="navigation-item "><span>Category 4</span></button>
        </div>
        <hr class="ssb-divider">
    </div>
</div>
```

#### React

```jsx harmony
<Header>
  {/* Fill with content, as below */}
    <div className="global-links" style={{ float: 'right', marginBottom: '12px', marginTop: '10px'  }}>
        <Link href=" ">top-item 3</Link>
        <Link href=" ">top-item 2</Link>
        <Link href=" ">top-item 1</Link>
    </div>
    <div className="top-row flex-row justify-space-between flex-wrap" style={{ width: '100%' }}>
        <SsbLogo alt="ssb-logo" />
        <div className="searchfield" style={{ width: '285px', alignSelf: 'flex-end' }}>
            <Input ariaLabel="Input field Search" searchField placeholder="Search text" />
        </div>
    </div>
    <div className="header-content" style={{ marginBottom: '20px', marginTop: '14px' }}>
        <Tabs items={items} />
        <Divider />
    </div>
</Header>

```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| children | node | All rendered content |
