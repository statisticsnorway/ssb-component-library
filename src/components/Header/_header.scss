.ssb-header-wrapper {
  box-sizing: border-box;
  padding: 0 114px;
  width: 100%;

  .global-links {

    > * {
      margin-left: 20px;
    }
  }
}
