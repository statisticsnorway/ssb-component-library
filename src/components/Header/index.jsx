import React from 'react';
import PropTypes from 'prop-types';

const Header = ({ children }) => (
	<div className="ssb-header-wrapper">
		{children}
	</div>
);

Header.defaultProps = {};

Header.propTypes = {
	children: PropTypes.node,
};

export default Header;
