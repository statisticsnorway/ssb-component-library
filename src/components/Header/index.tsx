import React, { type ReactNode } from 'react'

interface HeaderProps {
  children?: ReactNode
  className?: string
}

const Header: React.FC<HeaderProps> = ({ children, className }) => (
  <div className={`ssb-header-wrapper${className ? ` ${className}` : ''}`}>{children}</div>
)

export default Header
