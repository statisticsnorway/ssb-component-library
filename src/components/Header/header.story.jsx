import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import Header from './index';
import Tabs from '../Tabs';
import Divider from '../Divider';
import Link from '../Link';
import SsbLogo from '../../media/ssb-logo-black.svg';
import Input from '../Input';

const items = [
	{
		title: 'Category 1',
		path: 'category1/',
	},
	{
		title: 'Category 2',
		path: 'category2/',
	},
	{
		title: 'Category 3',
		path: 'category3/',
	},
	{
		title: 'Category 4',
		path: 'category4/',
	},
];

storiesOf('Header', module).addDecorator(centered)
	.add('Default', () => (
		<div style={{ width: '1200px' }}>
			<Header>
				<div className="global-links" style={{ float: 'right', marginBottom: '12px', marginTop: '10px' }}>
					<Link href=" ">top-item 3</Link>
					<Link href=" ">top-item 2</Link>
					<Link href=" ">top-item 1</Link>
				</div>
				<div className="top-row flex-row justify-space-between flex-wrap" style={{ width: '100%' }}>
					<SsbLogo alt="ssb-logo" />
					<div className="searchfield" style={{ width: '285px', alignSelf: 'flex-end' }}>
						<Input ariaLabel="Input field Search" searchField placeholder="Search text" />
					</div>
				</div>
				<div className="header-content" style={{ marginBottom: '20px', marginTop: '14px' }}>
					<Tabs items={items} />
					<Divider />
				</div>
			</Header>
		</div>
	));
