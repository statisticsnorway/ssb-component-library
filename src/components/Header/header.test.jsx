import React from 'react';
import { shallow } from 'enzyme';
import Header from './index';

describe('Header', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<Header><div>Content</div></Header>);
		expect(wrapper).toMatchSnapshot();
	});
});
