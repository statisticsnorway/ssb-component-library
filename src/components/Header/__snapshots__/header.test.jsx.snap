// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Matches the snapshot 1`] = `
<div
  className="ssb-header-wrapper"
>
  <div>
    Content
  </div>
</div>
`;
